app/graphql/types/query_type.rb

```diff
@@ -191,5 +189,64 @@ def query_files_whitelist(path, file_type)
       def query_file?(file, path, file_type)
         File.exist?(file) && query_files_whitelist(path, file_type).include?(file)
       end
+
+      def static_content_with_data_type(data_type, name, version = nil)
+        # HINT: mysql REGEXP for /^#{name}(-\d\.\d\.\d)?$/ was not working fully somehow,
+        #       so we query for all contents starting with name and filter with rails afterwards
+        static_contents = StaticContent.where("data_type = '#{data_type}' AND name LIKE '#{name}%'")
+
+        # filter static contents matching name and version number
+        #
+        # examples given the name "someName" and version "1.2.3":
+        #   someName -> true
+        #   someName-1.2.3 -> true
+        #   someName-new -> false
+        #   someName-new-1.2.3 -> false
+        static_contents = static_contents.filter do |static_content|
+          static_content.name.match(/^#{name}(-\d\.\d\.\d)?$/)
+        end
+
+        static_content = find_static_content(static_contents, name, version)
+
+        return { content: static_content.content, name: name } if static_content.present?
+
+        { content: data_type == "html" ? "" : {}, name: "not found" }
+      end
+
+      # loop through static contents for a queried name an optionally queried version
+      #
+      # @param [Array] static_contents all entries to loop through
+      # @param [String] query_name the content name that was queried
+      # @param [String] query_version an optional content version number that was queried
+      #
+      # @return [Object] a (closest) versioned or an unversioned static content
+      def find_static_content(static_contents, query_name, query_version = nil)
+        max_version = nil # refers to 0.0.0 with Gem::Version
+        versioned_static_content = nil
+        unversioned_static_content = nil
+
+        static_contents.each do |static_content|
+          regex_group = static_content.name.match(/#{query_name}-(.+)/)
+
+          unless regex_group
+            unversioned_static_content = static_content
+            next
+          end
+
+          version = regex_group[1]
+
+          next unless Gem::Version.new(max_version) < Gem::Version.new(version) &&
+                      Gem::Version.new(version) <= Gem::Version.new(query_version)
```

Comment:
@marcometz · 4 weeks ago

> Gem::Version.new(String) hier ist cool, macht natürlich dir Sortierung in SQL etwas schwieriger, oder? Eventuell würde das hier gehen: https://stackoverflow.com/a/8408337/1230300

Reply:
what do you mean with more difficult sorting in SQL? how is that related here?

Reply:
ich dachte du möchtest vielleicht alle StaticContents bereits sortiert nach der Version aus der DB laden aber derzeit passiert der Vergleich ja in Ruby.

Frage:

`[SC:0.9.0, SC:1.0.1, SC:1.0.2, SC:1.0.0]`

Wenn du jetzt bsp. auf der Suche nach Version 1.0.0 bist (`query_version = 1.0.0`)

1. Durchlauf 0.9.0 < 1.0.0 => next
2. Durchlauf 1.0.1 > 1.0.0 => return 1.0.1

SC:1.0.0 wird dann glaube nicht gefunden.
`static_contents = StaticContent.where(data_type: data_type, name: name)` braucht glaube ein `order|sort` nach der Version

Reply:
auf datenbankebene wüsste ich nicht, wie versionsnummern verglichen werden können. sind ja nicht einfach nur zahlen. dafür verwenden wir ja extra das gem. 

das mit dem durchlauf kann ich grad gar nicht beantworten genau. müsste ich wieder tiefer einsteigen. aber ich probier dein beispiel mal nachzustellen in meiner lokalen datenbank bzw. in einem spec und sag dir dann was rauskam :)

Reply:
mit meinen tests war allet tutti, also: ja